engines/router/missionctl/handlers/batch_http_handler.go

```diff
@@ -0,0 +1,124 @@
+package handlers
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"sync"
+
+	"github.com/gojek/turing/engines/router/missionctl"
+	"github.com/gojek/turing/engines/router/missionctl/errors"
+	"github.com/gojek/turing/engines/router/missionctl/log"
+	"github.com/gojek/turing/engines/router/missionctl/turingctx"
+)
+
+type batchHTTPHandler struct {
+	httpHandler
+}
+
+type batchResponse struct {
+	StatusCode int         `json:"code"`
+	ErrorMsg   string      `json:"error,omitempty"`
+	Data       interface{} `json:"data,omitempty"`
+}
+
+type batchResult struct {
+	BatchResult []batchResponse `json:"batch_result"`
+}
+
+// NewBatchHTTPHandler creates an instance of the Mission Control's prediction request handler
+func NewBatchHTTPHandler(mc missionctl.MissionControl) http.Handler {
+	return &batchHTTPHandler{httpHandler{mc}}
+}
+
+func (h *batchHTTPHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
+	var httpErr *errors.HTTPError
+	h.measureRequestDuration(httpErr)
+
+	// Create context from the request context
+	ctx := turingctx.NewTuringContext(req.Context())
+	// Create context logger
+	ctxLogger := log.WithContext(ctx)
+	defer func() {
+		_ = ctxLogger.Sync()
+	}()
+
+	// Get the unique turing request id from the context
+	turingReqID, err := turingctx.GetRequestID(ctx)
```

Comment:
Are there any side effects if we have the same turing request ID for multiple predictions? Would it mess with result logging? @krithika369 can you think of any?

Reply:
I can't think of any. There is no uniqueness constraint on the Turing Request ID. However, I'm curious - is that the only distinguisher in the logs - i.e., the presence of multiple records with the same ID indicate it's a batch request? 

Should we instead append a suffix to the TuringReqID? So, for batch requests, they will be `{unique_hash}-{idx}` where the idx is their index in the original request. That way, we have a common pattern and a distinguisher for batch requests. wdyt?

Reply:
added it, will look something like this
![Screenshot 2021-11-25 at 9 20 13 AM](https://user-images.githubusercontent.com/30390872/143336835-27b25d9c-0bd0-4574-b90b-a5d58dc334df.png)